skbio/io/format/sample_metadata.py

```diff
@@ -0,0 +1,93 @@
+"""Sample Metadata object ported over from qiime2.
+
+===============================================
+
+.. currentmodule:: skbio.io.format.sample_metadata
+
+Format Support
+--------------
+**Has Sniffer: Yes**
+
++------+------+---------------------------------------------------------------+
+|Reader|Writer|                          Object Class                         |
++======+======+===============================================================+
+|Yes   |Yes   |:mod:`skbio.metadata.SampleMetadata`                              |
++------+------+---------------------------------------------------------------+
+"""
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import csv
+import re
+
+from skbio.io import create_format
+from skbio.metadata._metadata import SampleMetadata
+from skbio.metadata.io import MetadataReader, MetadataWriter
+
+
+sample_metadata = create_format("sample_metadata")
+
+
+@sample_metadata.sniffer()
+def _sample_metadata_sniffer(fh):
+    # Strategy:
+    # Check if first word is in the file is in the list
+    # of allowed metadata words
+    try:
+        tsv_reader = csv.reader(fh, dialect="excel-tab", strict=True)
+        # sample id and feature id are not separated when reading the tsv
+        # since they are not tab-separated.
+        possible_ids = [
```

Comment:
It did not work with #SampleID however I modified the sniffer so that it now correctly identifies files with #SampleID as described in the qiime2 documentation.